### ISSUE TITLE
Coordination Meeting 2025

### ISSUE THREAD

**Issue by pllim** [astropy/astropy-project#430]
# Will lock issue on 2025-05-09

Like #349 but for 2025 meeting.

[2025 meeting wiki](https://github.com/astropy/astropy/wiki/UCSC-Meeting-2025)

[Running notes from 2024 meeting](https://docs.google.com/document/d/1XFtLOgbdJTBo8n2_5fUCP1UJdYuJV8KklK7oVUdmaVE/view)

[Project roadmap](https://github.com/astropy/astropy-project/blob/main/roadmap/roadmap.md)

**Comment by pllim** [comment]
If we try something from https://astropy-dei.orgmycology.com/ after Coordination Meeting 2024, should report back on what we tried and whether it worked.

**Comment by pllim** [comment]
Array API + Quantity 2.0 -- Lots of discussions at Coordination Meeting 2024. Any progress after a year?

**Comment by pllim** [comment]
Did we implement anything discussed in the [Are there big missing pieces?](https://docs.google.com/document/d/1lOUQOXLVjZVDPImfUpumFos4Abw_aQQoI1-mDHS1wgU/edit?usp=sharing) breakout at Coordination Meeting 2024?

**Comment by pllim** [comment]
Annual roadmap discussions.

**Comment by pllim** [comment]
Strategic Planning Role -- If we ended up filling the role after Coordination Meeting 2024, did it work?

**Comment by pllim** [comment]
Note for SOC: If the LOC plans to do a press release about this meeting, should add a "are you okay with your photo being used" question in the registration form. Thanks!

**Comment by pllim** [comment]
APE 22 follow-up (if necessary):

* https://github.com/astropy/astropy.github.com/issues/571
* https://github.com/astropy/astropy-project/issues/437

**Comment by pllim** [comment]
Reports from projects that got money from Astropy since the last Coordination Meeting (Cycle 4?).

**Comment by pllim** [comment]
Celebrate getting 2025 Lancelot M. Berkeley–New York Community Trust Prize for Meritorious Work in Astronomy from AAS.

Press releases:

* https://aas.org/press/astropy-collaboration-receive-2025-berkeley-prize
* https://www.stsci.edu/contents/news-releases/2024/news-2024-401

**Comment by adrn** [comment]
In addition to the [array API bullet](https://github.com/astropy/astropy-project/issues/430#issuecomment-2183074347), it would be useful to resurrect the discussion of creating a "Quantity/unit API" analog to the array API (cc @mhvk @nstarman @jeffjennings)

**Comment by pllim** [comment]
Re: https://github.com/astropy/astropy-project/issues/430#issuecomment-2391735657

@adrn -- Is that already an issue somewhere?

**Comment by adrn** [comment]
I don't know if it's an issue or an email thread or a section of a google doc or a discussion in a separate issue, but @mhvk or @nstarman might remember :).

**Comment by mhvk** [comment]
Discussion so far has been mostly about Quantity 2.0, at https://github.com/astropy/astropy-APEs/pull/91 - there hasn't been much discussion about a units API beyond https://github.com/astropy/astropy/issues/8210

**Comment by pllim** [comment]
@astrofrog -- Hack session to reconsider layout of current astropy documentation landing page.

**Comment by pllim** [comment]
Leverage https://github.com/scientific-python/sync-teams-action to automate project team membership and access? Would like to discuss pros, cons, security implications specific to our project.

**Comment by nstarman** [comment]
Switching to Meson build system?
My primary motivation is that it would be incredible if Astropy could automate the release process to the point that bugfix releases could be done on a small-n group of bugs basis.

**Comment by nstarman** [comment]
Idea 2: Hack day using LLMs to rewrite some of our various C(ython,++,#,?) codes into a single backend, probably whichever best supports Numpy.

**Comment by nstarman** [comment]
Idea 3) Build a feature release & planned refactors GH Project board. Then we can see what everyone in Astropy thinks needs improving / adding to.

**Comment by Cadair** [comment]
> My primary motivation is that it would be incredible if Astropy could automate the release process to the point that bugfix releases could be done on a small-n group of bugs basis.

I am +1 on this, but I don't think it would involve swapping build system, we could almost certainly automate most of the drudgery with actions.

**Comment by pllim** [comment]
Re: automate release -- What exactly needs to be automated still? We already use the publish workflow from OpenAstronomy. Things like website update and community announcement are still manual but a matter of copy paste.

**Comment by pllim** [comment]
@dhomeier -- Update code recommendation for Unicode

* https://github.com/astropy/astropy/issues/17731

**Comment by eli-schwartz** [comment]
> Switching to Meson build system?
> My primary motivation is that it would be incredible if Astropy could automate the release process to the point that bugfix releases could be done on a small-n group of bugs basis.
> Also, match scipy and other leaders of the ecosystem.

The main advantage that SciPy sought from switching to meson wasn't about automating the release process but rather more reliable and speedier builds :) especially because setuptools is effectively impossible to really parallelize whereas meson is extremely parallelizable.

**Comment by pllim** [comment]
New policy discussion -- removing roles

* https://github.com/astropy/astropy-project/issues/462

**Comment by astrofrog** [comment]
> Re: automate release -- What exactly needs to be automated still?

I mean in principle bugfix releases could be *completely* automatic and happen periodically if there are new backports to the release branch

**Comment by pllim** [comment]
### astropy user survey

Action item for the meeting: @jeffjennings (or his colleagues involved in the survey) could present survey results.

As posted on https://groups.google.com/g/astropy-dev/c/5JqGiadcufw

**Example announcement:**

The Astropy Project is running its first [user survey](https://forms.gle/KX2cNuFsQYgVhF178)! Please take 15 minutes to give your input before April 1 to help us better understand how astropy is being used, how it can be improved, and how we can best continue to support the astronomy and scientific research communities. Thank you in advance for helping us shape the future of Astropy!

**Comment by kbwestfall** [comment]
I've gotten some responses regarding possible tutorial/hack sessions, specifically from the group community faculty that participate in our [ASTRAL program](https://www.ucobservatories.org/education/astral-astronomy-stem-alliance-with-lick-observatory/):

- point-source aperture photometry
- creating color images from individual images in 3 filters
- image stacking and/or creating continuum-subtracted narrowband images

**Comment by pllim** [comment]
Re: https://github.com/astropy/astropy-project/issues/430#issuecomment-2702208314 -- I think @mwcraig might already have materials for these but maybe he can confirm.

**Comment by neutrinoceros** [comment]
I would like to propose discussions on:
- sustainability of our testing strategies (https://github.com/astropy/astropy/issues/17596)
- modernizing our build system (https://github.com/astropy/astropy/issues/17760)

I've studied both topics in enough depth that I could do a presentation for either, and I think both would benefit from project-level discussions.

**Comment by neutrinoceros** [comment]
Also, and perhaprs more importantly: free-threading Python is coming for us
- https://github.com/astropy/astropy/issues/16916

(I can also give a talk on this)

**Comment by pllim** [comment]
Astropy Coordination Meeting 2026 -- Early volunteers for SOC next year and potential hosts.

**Comment by jeffjennings** [comment]
[APE 26](https://github.com/astropy/astropy-APEs/pull/112) debrief: summary of proposal, discussion and next steps @adrn @nstarman @mhvk

**Comment by ceb8** [comment]
Discussion about changing the requirements for changing APE 0.

**Comment by tepickering** [comment]
there was a discussion on slack about `astropy.utils.data`, it's applicability outside of core `astropy`, and other similar tools out there. might be worth picking that up in person and maybe whip up the suggested documentation PR.

**Comment by astrofrog** [comment]
I'd be interested in having a session to talk about current best approaches for leveraging GPUs (currently an item in the roadmap [here](https://github.com/astropy/astropy-project/blob/main/roadmap/roadmap.md#hardware-and-performance)) - for example how can we make use of packages such as [jax](https://docs.jax.dev/en/latest/quickstart.html) or other similar packages in astropy?

**Comment by pllim** [comment]
SOC will now take the ideas and figure out a tentative schedule. We will no longer be accepting new suggestions. There is no option to lock this issue for everyone though, so 🤷‍♀ ... Thanks for all the contributions!

**Comment by pllim** [comment]
p.s. I think we need a better way to collect this stuff if we want to keep hybrid because of the time zone info needed also for scheduling. 🤪
